src/Apps/Order/Routes/Shipping2/SavedAddresses2.tsx

```diff
@@ -0,0 +1,276 @@
+import * as React from "react"
+import { useCallback, useState } from "react"
+import styled from "styled-components"
+import { compact } from "lodash"
+import { RadioGroup, BorderedRadio, Spacer, Clickable } from "@artsy/palette"
+import { createRefetchContainer, graphql, RelayRefetchProp } from "react-relay"
+import { SavedAddresses2_me$data } from "__generated__/SavedAddresses2_me.graphql"
+import {
+  AddressModal,
+  AddressModalAction,
+  AddressModalActionType,
+} from "Apps/Order/Routes/Shipping2/AddressModal2"
+import createLogger from "Utils/logger"
+import { SavedAddressItem } from "Apps/Order/Routes/Shipping2/SavedAddressItem2"
+import { extractNodes } from "Utils/extractNodes"
+import { useTracking } from "react-tracking"
+import {
+  ActionType,
+  ClickedShippingAddress,
+  ContextModule,
+  OwnerType,
+} from "@artsy/cohesion"
+import { themeGet } from "@styled-system/theme-get"
+
+import { useShippingContext } from "Apps/Order/Routes/Shipping2/ShippingContext"
+import {
+  SavedAddressType,
+  ShippingAddressFormValues,
+  addressWithFallbackValues,
+  getDefaultUserAddress,
+} from "Apps/Order/Routes/Shipping2/shippingUtils"
+
+export const NEW_ADDRESS = "NEW_ADDRESS"
+const PAGE_SIZE = 30
+
+export interface SavedAddressesProps {
+  relay: RelayRefetchProp
+  me: SavedAddresses2_me$data
+  active: boolean
+  onSelect: (address: ShippingAddressFormValues) => void
+}
+
+const getAddressByID = (addressList: SavedAddressType[], addressID: string) => {
+  return addressList.find(node => node.internalID === addressID)
+}
+
+const getBestAvailableAddress = (
+  addressList: SavedAddressType[],
+  addressID?: string,
+  availableShippingCountries?: string[]
+) => {
+  return (
+    (addressID && getAddressByID(addressList, addressID)) ||
+    getDefaultUserAddress(addressList, availableShippingCountries)
+  )
+}
```

Comment:
#superminor - can these helper functions be moved below the primary concern (`const SavedAddress = ... `). Its nice to not have to wade through intermediate stuff before getting to the part that's most likely to change in the future.

Reply:
Or, since there's already a shippingUtils file, it can just go there

Reply:
Moved to the bottom for now. I'm hesitant to use the shippingUtils file for things that aren't shared across multiple components at this point but open to leaning on it more later or others making that decision.